BSP/Simulator/Src/BSP_UART.c

```diff
@@ -50,104 +64,141 @@ void BSP_UART_Init(void) {
  * @pre     str should be at least 128bytes long.
  * @param   str : pointer to buffer to store the string. This buffer should be initialized
  *                  before hand.
+ * @param   usart : which usart to read from
  * @return  number of bytes that was read
  */
-uint32_t BSP_UART_ReadLine(char *str) {
-    if(lineReceived) {
-        pthread_mutex_lock(&rx_mutex);
-        uint8_t data = 0;
-        uint32_t recvd = 0;
-        RxFifo_Peek(&data);
-        while(!RxFifo_IsEmpty() && data != '\r' && data != '\n') {
-            recvd += RxFifo_Get((uint8_t *)str++);
-            RxFifo_Peek(&data);
+uint32_t BSP_UART_ReadLine(char *str, UART_Port usart) {
+    uint8_t data = 0;
+    uint32_t recvd = 0;
+    if(lineReceived3 && (usart == UART_USB)) {
+        pthread_mutex_lock(&rx_mutex3);
+        RxFifo_Peek(&data, UART_USB);
+        while(!RxFifo_IsEmpty(UART_USB) && data != '\r' && data != '\n') {
+            recvd += RxFifo_Get((uint8_t *)str++, UART_USB);
+            RxFifo_Peek(&data, UART_USB);
         }
-
-        RxFifo_Get(&data);
-
+        RxFifo_Get(&data, UART_USB);
         *str = 0;
-
-        lineReceived = false;
-
-        pthread_mutex_unlock(&rx_mutex);
+        lineReceived3 = false;
+        pthread_mutex_unlock(&rx_mutex3);
+        return recvd;
+    }
+    if(lineReceived2 && (usart == UART_BLE)) {
+        pthread_mutex_lock(&rx_mutex2); //NEED TO CHANGE THIS **********
```

Comment:
Do we still need to change this?

Reply:
No, I left that there in an earlier version of my code.